### PR TITLE
Refactor soci-store code out of podman.go and support restarting the store at a different mountpoint

### PR DESCRIPTION
**Pull request by iain-macdonald**
I wrote a PR that just added the new feature, but I've been unhappy with how intertwined the soci-store and podman code are for a while, so I figured it's probably time to bite the bullet and actually make the interaction between these two more sane.

This PR pulls all of the code for starting up, managing, and interacting with the soci-store out of `podman.go` and puts it in a new `soci-store` package which exports three versions of an interface abstracting the soci-store functionality away:
* A `NoStore` which does not start a soci-store and tells podman not to use one.
* A `DevStore` for cases where one wants to run the soci-store as a separate process. This is useful if you want to do stuff like make it crash, or add extra logging, though I intend to get rid of it.
* A `SociStore` that is the real deal.

**Related issues**: N/A

### CODE REVIEW

File: enterprise/server/remote_execution/containers/soci_store/soci_store.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/util/oci"
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
+)
+
+// An abstraction of the soci-store that enables podman image streaming.
+type Store interface {
+	// Returns nil if the soci-store is alive and ready to serve traffic, or an
+	// error if not.
+	Exists(ctx context.Context) error
```
[the remark this answers is not in the document]
Comment (iain-macdonald) [reply enterprise/server/remote_execution/containers/soci_store/soci_store.go]: Done

File: enterprise/server/remote_execution/containers/soci_store/soci_store.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+	// Waits until the store is alive and ready to serve traffic, returning
+	// an error if it takes too lnog.
+	WaitUntilExists() error
+
+	// Returns the command-line argument to pass to podman in order to stream
+	// images using this soci-store.
+	EnableStreamingStoreArg() string
```
[the remark this answers is not in the document]
Comment (iain-macdonald) [reply enterprise/server/remote_execution/containers/soci_store/soci_store.go]: Done

File: enterprise/server/remote_execution/containers/soci_store/soci_store_linux.go [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+type SociStore struct {
+	rootPath            string
+	artifactStoreClient socipb.SociArtifactStoreClient
+	keychainClient      sspb.LocalKeychainClient
+
+	idx *int
+	mu  *sync.RWMutex
+}
+
+func (s SociStore) Exists(ctx context.Context) error {
```
[the remark this answers is not in the document]
Comment (iain-macdonald) [reply enterprise/server/remote_execution/containers/soci_store/soci_store_linux.go]: Done

File: enterprise/server/remote_execution/containers/soci_store/soci_store.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
+package soci_store
```
Comment (bduffany) [inline enterprise/server/remote_execution/containers/soci_store/soci_store.go:1]: nit: the `containers` dir currently contains one package per supported isolation type. Seems like this pkg might fit better directly under remote_execution/?

**Review by bduffany — APPROVED** [review]

**Pull request merged**